### PR TITLE
Cleaning up XHR functions, refactored output processing

### PR DESCRIPTION
**Pull request by imlvts**
- ajax functions no longer rely on the state of DOM
- `librest` and `libdom` were merged to `libs.js`
- printing no longer requires `innerHTML` (except for `scani2c.html`)
- fixed several lint errors - missing var declarations, semicolons, etc.

### CODE REVIEW

File: firmware-src/pages/gpio.html [hunk 1]
```diff
@@ -2,39 +2,31 @@
...
-			document.getElementById('output').innerHTML = text;
-		}
-
-		function send(command, data, cb) {
-			print('');
-			rest_common(command, data, function (err, res) {
+	<script src="libs.js"></script>
+	<script>
+		function get(pin) {
+			send_command('gpio/read', data, getKey(), function (err, res) {
```
Comment (Nikolay-Kha) [inline firmware-src/pages/gpio.html]: Uncaught ReferenceError: data is not defined
    at get (gpio.html:8)

File: firmware-src/pages/pwm.html [hunk 2]
```diff
@@ -2,31 +2,26 @@
...
+		function setPwm(dc, count, frequency) {
 			print('');
-			var params = {}
-			params[document.getElementById('pin').value] = dc;
+			var params = {};
+			params[byId('pin').value] = dc;
 			params['frequency'] = frequency;
 			params['count'] = count;
-			rest_common('pwm/control', params, function (err, res) {
+			send_command('pwm/control', params, function (err, res) {
```
Comment (Nikolay-Kha) [inline firmware-src/pages/pwm.html]: Uncaught DOMException: Failed to execute 'setRequestHeader' on 'XMLHttpRequest': 'Bearer function (err, res) {
				if(err)
					print('ERROR: ' + res, true);
			}' is not a valid HTTP header field value.
    at ajax (http://esp-test.vrn.dataart.net/libs.js:5:10)
    at send_command (http://esp-test.vrn.dataart.net/libs.js:22:2)
    at setPwm (http://esp-test.vrn.dataart.net/pwm.html:13:4)
    at HTMLButtonElement.onclick (http://esp-test.vrn.dataart.net/pwm.html:45:85)

File: firmware-src/pages/tryapi.html [hunk 3]
```diff
@@ -2,23 +2,21 @@
...
-		function printinput(text) {
-			document.getElementById('input').innerHTML = text;
-		}
-
-		function print(text, err = false) {
-			document.getElementById('output').style.color = err ? 'red' : 'green';
-			document.getElementById('output').innerHTML = text;
+	<script src="libs.js"></script>
+	<style>
+	#output, #input pre { white-space: pre-wrap }
```
Comment (Nikolay-Kha) [inline firmware-src/pages/tryapi.html:7]: May be it should be moved to style.css?

Comment (imlvts) [reply firmware-src/pages/tryapi.html:7]: I don't think so. `tryapi.html` is the only document that needs to reliably display multiline data. This style could be removed, but that would add some complexity to the rendering code.

Comment (Nikolay-Kha) [reply firmware-src/pages/tryapi.html:7]: ok

**Pull request merged**